src/backend/crossterm.rs

```diff
@@ -127,6 +145,223 @@ where
     }
 }
 
+impl CrosstermBackend<io::Stdout> {
+    /// Creates a new `CrosstermBackend` with `std::io::stdout`, enables raw mode, and switches to
+    /// the alternate screen.
+    ///
+    /// Raw mode and alternate screen are restored when the `CrosstermBackend` is dropped.
+    ///
+    /// # Example
+    ///
+    /// ```rust,no_run
+    /// # use ratatui::backend::CrosstermBackend;
+    /// let backend = CrosstermBackend::stdout();
+    /// ```
+    pub fn stdout() -> io::Result<Self> {
+        Self::new(io::stdout())
+            .with_raw_mode()?
+            .with_alternate_screen()
+    }
```

Comment:
The PR description says:

> - `CrosstermBackend::stdout` and `CrosstermBackend::stderr` for creating backends with std::io::stdout and std::io::stderr respectively.
> - `CrosstermBackend::stdout_with_defaults` and `CrosstermBackend::stderr_with_defaults` for creating backends with that use the alternate screen and raw mode.

which doesn't match this implementation. 

I prefer it to work the way the PR description mentions. It makes it much more clearer imo.

Reply:
Alternatively, `CrosstermBackend::stdout_tui` and `CrosstermBackend::stderr_tui`.

Reply:
The idea behind doing some reasonable defaults (raw/alternate screen) on these is that this is necessary for 90% of all apps. Anything else can do `CrosstermBackend::new(stdout()).with_xxx()`...

Reply:
It's a little confusing imo to find in a Ratatui app's user code two things that are very similar but do different things

- `CrosstermBackend::new(stdout())` for inline
- `CrosstermBackend::stdout()` for a TUI

Reply:
I like the reasonable defaults in place here, I'm just not convinced that the name conveys those defaults clearly.

Reply:
I like the idea of `CrosstermBackend::stdout_tui()` with the alternative of `CrossteermBackend::new(stdout()).with_*` is actually quite good. It removes the `CrosstermBackend::stdout()` == `CrosstermBackend::new(stdout())` similarity and is not just `default` which is more confusing as there are different defaults for different use cases.